pkg/controller/scylladbmonitoring/sync_grafana.go

```diff
@@ -140,9 +140,25 @@ func makeGrafanaConfigs(sm *scyllav1alpha1.ScyllaDBMonitoring) (*corev1.ConfigMa
 }
 
 func makeGrafanaDashboards(sm *scyllav1alpha1.ScyllaDBMonitoring) (*corev1.ConfigMap, string, error) {
-	return grafanav1alpha1assets.GrafanaDashboardsConfigMapTemplate.RenderObject(map[string]any{
-		"scyllaDBMonitoringName": sm.Name,
-	})
+	smType := sm.Spec.Type
+	if smType == nil {
+		// This should have a default value, but it can be nil due to a version skew.
+		scyllaDBMonitoringTypeSAAS := scyllav1alpha1.ScyllaDBMonitoringTypeSAAS
+		smType = &scyllaDBMonitoringTypeSAAS
+	}
+
+	switch t := *smType; t {
```

Comment:
there is no need for this to be a pointer
```suggestion
	t := scyllav1alpha1.ScyllaDBMonitoringTypeSAAS
	if sm.Spec.Type != nil {
	    t = *sm.Spec.Type
	}
	
	
	switch t {
```